### PR TITLE
Add route to retrieve a purl with a version ID

### LINKED ISSUE #1039: Update display to use versioned metadata
With versioned metadata available, the purl display will need to be updated to use the metadata for the version provided, or the latest version if not provided.

Add an optional version parameter to the purl URL that will be used for direct linking into a version and citations with version information.

### PR DESCRIPTION
**Pull request by mjgiarlo**
Fixes #1039

The only UI change is this banner when viewing any version other than the head version:

Screenshot from 2024-07-24 14-06-04

This does not implement the UI as specified in the Figma. Instead, it uses the existing code for handling alerts, for consistency reasons.

### CODE REVIEW

File: app/controllers/purl_controller.rb [hunk 1]
```diff
@@ -12,9 +12,11 @@ def index; end
   # rubocop:disable Metrics/AbcSize
   # rubocop:disable Metrics/MethodLength
   def show
-    @version = @purl.version(:head)
+    @version = @purl.version(version_param)
     return unless stale?(last_modified: @version.updated_at.utc, etag: @version.cache_key + "/#{@version.updated_at.utc}")
 
+    flash[:alert] = 'A newer version of this item is available' unless @version.head?
```
Comment (cbeer) [inline app/controllers/purl_controller.rb:18]:
```suggestion
    flash.now[:alert] = 'A newer version of this item is available' unless @version.head?
```

I think?

File: app/controllers/purl_controller.rb [hunk 2]
```diff
@@ -12,9 +12,11 @@ def index; end
   # rubocop:disable Metrics/AbcSize
   # rubocop:disable Metrics/MethodLength
   def show
-    @version = @purl.version(:head)
+    @version = @purl.version(version_param)
```
Comment (cbeer) [inline app/controllers/purl_controller.rb:15]: What should we do if the version doesn't exist? (Presumably not a 500 error: https://app.honeybadger.io/projects/49629/faults/109769206)

Comment (mjgiarlo) [reply app/controllers/purl_controller.rb:15]: I'll whip up a test for this and handle the error.

File: app/controllers/purl_controller.rb (hunk 1 above)
Comment (mjgiarlo) [reply app/controllers/purl_controller.rb:18]: This is what Rubocop wanted, but I didn't see any usage of `flash.now` in purl so I was hesitant to patch that in. Safe enough to try and test it.

Comment (cbeer) [reply app/controllers/purl_controller.rb:18]: Huh. I thought `flash[]` was for redirects, and `flash.now` was for rendering. Maybe that's not cool any more 🤷‍♂️

Comment (mjgiarlo) [reply app/controllers/purl_controller.rb:18]: @cbeer the request spec is happy with the change at least.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**